### PR TITLE
WriteCombinedReference returns `zarr.storage.FSStore`

### PR DESCRIPTION
**Pull request by norlandrhagen**
PR to address Issue #608

### CODE REVIEW

File: pangeo_forge_recipes/writers.py [hunk 1]
```diff
@@ -148,7 +148,14 @@ def write_combined_reference(
     else:
         raise NotImplementedError(f"{file_ext = } not supported.")
 
-    return full_target
+    # Return an fsspec mapper that can be read with Xarray
+    return ReferenceFileSystem(
+        outpath,
+        remote_protocol=remote_protocol,
+        remote_options={"anon": True},
+        target_protocol="file",
```
Comment (cisaacstern) [inline pangeo_forge_recipes/writers.py:156]: Comments on each of these:

- `remote_protocol`: IIUC, this is the protocol the data lives on, which is not what this variable has been assigned to above. This variable is assigned to to the protocol that the reference is written to.
- `remote_options`: We don't know for sure that these are the options users will need to access the remote data. This needs to be configurable as a kwarg. Possibly `anon:True` is a sensible default for that kwarg.
- `target_protocol`: IIUC, this is the protocol on which the reference file exists. So it is the the value that the `remote_protocol` variable is currently assigned to.

File: pangeo_forge_recipes/writers.py [hunk 2]
```diff
@@ -112,13 +112,15 @@ def write_combined_reference(
 ) -> FSSpecTarget:
     """Write a kerchunk combined references object to file."""
 
-    import ujson  # type: ignore
+    from fsspec.implementations.reference import ReferenceFileSystem
 
     file_ext = os.path.splitext(output_file_name)[-1]
-
     outpath = full_target._full_path(output_file_name)
+    remote_protocol = _select_single_protocol(full_target)
```
Comment (cisaacstern) [inline pangeo_forge_recipes/writers.py:119]:
```suggestion
    target_protocol = _select_single_protocol(full_target)
```

See other comment re: difference between `remote_protocol` and `target_protocol`. IIUC, we have these backwards at the moment.

**Review by cisaacstern — COMMENTED** [review]
Getting close, @norlandrhagen.

A few comments below re: generalizing the arguments to ReferenceFileSystem.

File: pangeo_forge_recipes/writers.py (hunk 2 above)
Comment (norlandrhagen) [reply pangeo_forge_recipes/writers.py:119]: So where/how can the remote protocol be inferred? Or should it be passed explicitly? 

Also, should all the the kwargs that we pass to `MultiZarrToZarr` and `ReferenceFileSystem` be packaged as mzz_kwargs and unpacked or separated?

**Comment by norlandrhagen** [comment]
@cisaacstern good call!

Yeah, it seems like `remote_protocol` should be inferable from the `MultiZarrToZarr` object. 

I just pushed a commit that specifies the `remote_protocol` == 'file' upstream in `OpenWithKerchunk`. However in `write_combined_references` the `MZZ` object `reference.remote_protocol` returns None? So maybe the remote protocol is getting dropped somewhere or I'm missing something.

**Comment by cisaacstern** [comment]
@norlandrhagen the `reference` received by `write_combine_references` is emitted by the [`CombineMultiZarrToZarr` here](https://github.com/norlandrhagen/pangeo-forge-recipes/blob/72b02c6e32c07b0529f628726a9a7739b6b876f2/pangeo_forge_recipes/transforms.py#L440C13-L446).

I suspect that the issue may have something to do with the (lack of?) remote protocol being passed explicitly (or otherwise handled) there.

The difficulties of implementing this do make me think it may be worth pausing to implement https://github.com/pangeo-forge/pangeo-forge-recipes/issues/623 first, which would give us a much better testbed.

File: pangeo_forge_recipes/writers.py [hunk 3]
```diff
@@ -108,17 +108,21 @@ def write_combined_reference(
     full_target: FSSpecTarget,
     concat_dims: List[str],
     output_file_name: str,
+    # remote_protocol:  Optional[str],
+    remote_options: Optional[dict] = {"anon": True},
     refs_per_component: int = 1000,
 ) -> FSSpecTarget:
```
Comment (sbquinlan) [inline pangeo_forge_recipes/writers.py:114]: nit: update return typehint

**Comment by norlandrhagen** [comment]
Closing this in favor of PR: https://github.com/pangeo-forge/pangeo-forge-recipes/pull/660

**Pull request closed without merging**